### ISSUE TITLE
Container image cluster-api-azure-controller:v1.3.1 is not available

### ISSUE THREAD

**Issue by ionutbalutoiu** [kubernetes-sigs/cluster-api-provider-azure#2314]
/kind bug

Container image `cluster-api-azure-controller` for release `v1.3.1` is not available.

**What steps did you take and what happened:**

```
$ clusterctl init --infrastructure azure --wait-providers

Fetching providers
Installing cert-manager Version="v1.5.3"
Waiting for cert-manager to be available...
Installing Provider="cluster-api" Version="v1.1.3" TargetNamespace="capi-system"
Installing Provider="bootstrap-kubeadm" Version="v1.1.3" TargetNamespace="capi-kubeadm-bootstrap-system"
Installing Provider="control-plane-kubeadm" Version="v1.1.3" TargetNamespace="capi-kubeadm-control-plane-system"
Installing Provider="infrastructure-azure" Version="v1.3.1" TargetNamespace="capz-system"
Waiting for providers to be available...
Error: timed out waiting for the condition
```

Inspecting the `capz-system` pods, I noticed:
```
$ kubectl get pods -n capz-system

NAME                                       READY   STATUS             RESTARTS   AGE
capz-controller-manager-654ccfb847-xh762   0/1     ImagePullBackOff   0          31m
capz-nmi-wtd4p                             1/1     Running            0          31m
```

and describing the errored pod:

```
$ kubectl describe pod -n capz-system capz-controller-manager-654ccfb847-xh762

Name:         capz-controller-manager-654ccfb847-xh762
Namespace:    capz-system
Priority:     0
Node:         kind-control-plane/172.18.0.2
Start Time:   Thu, 19 May 2022 12:49:35 +0000
Labels:       aadpodidbinding=capz-controller-aadpodidentity-selector
              cluster.x-k8s.io/provider=infrastructure-azure
              control-plane=capz-controller-manager
              pod-template-hash=654ccfb847
Annotations:  kubectl.kubernetes.io/default-logs-container: manager
Status:       Pending
IP:           10.244.0.11
IPs:
  IP:           10.244.0.11
Controlled By:  ReplicaSet/capz-controller-manager-654ccfb847
Containers:
  manager:
    Container ID:
    Image:         us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1
    Image ID:
    Ports:         9443/TCP, 9440/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      --leader-elect
      --metrics-bind-addr=localhost:8080
      --feature-gates=MachinePool=false,AKS=false
      --v=0
    State:          Waiting
      Reason:       ImagePullBackOff
    Ready:          False
    Restart Count:  0
    Liveness:       http-get http://:healthz/healthz delay=10s timeout=1s period=10s #success=1 #failure=3
    Readiness:      http-get http://:healthz/readyz delay=10s timeout=1s period=10s #success=1 #failure=3
    Environment:
      AZURE_SUBSCRIPTION_ID:  <set to the key 'subscription-id' in secret 'capz-manager-bootstrap-credentials'>  Optional: false
      AZURE_TENANT_ID:        <set to the key 'tenant-id' in secret 'capz-manager-bootstrap-credentials'>        Optional: false
      AZURE_CLIENT_ID:        <set to the key 'client-id' in secret 'capz-manager-bootstrap-credentials'>        Optional: false
      AZURE_CLIENT_SECRET:    <set to the key 'client-secret' in secret 'capz-manager-bootstrap-credentials'>    Optional: false
      NODE_NAME:               (v1:spec.nodeName)
      POD_NAME:               capz-controller-manager-654ccfb847-xh762 (v1:metadata.name)
      POD_NAMESPACE:          capz-system (v1:metadata.namespace)
    Mounts:
      /tmp/k8s-webhook-server/serving-certs from cert (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-pksgs (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True
Volumes:
  cert:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  capz-webhook-service-cert
    Optional:    false
  kube-api-access-pksgs:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason       Age                    From               Message
  ----     ------       ----                   ----               -------
  Normal   Scheduled    32m                    default-scheduler  Successfully assigned capz-system/capz-controller-manager-654ccfb847-xh762 to kind-control-plane
  Warning  FailedMount  32m (x3 over 32m)      kubelet            MountVolume.SetUp failed for volume "cert" : secret "capz-webhook-service-cert" not found
  Warning  Failed       30m (x5 over 32m)      kubelet            Error: ImagePullBackOff
  Normal   Pulling      30m (x4 over 32m)      kubelet            Pulling image "us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1"
  Warning  Failed       30m (x4 over 32m)      kubelet            Failed to pull image "us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1": rpc error: code = NotFound desc = failed to pull and unpack image "us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1": failed to resolve reference "us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1": us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1: not found
  Warning  Failed       30m (x4 over 32m)      kubelet            Error: ErrImagePull
  Normal   BackOff      2m17s (x131 over 32m)  kubelet            Back-off pulling image "us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1"
```

**What did you expect to happen:**

Container iamge `us.gcr.io/k8s-artifacts-prod/cluster-api-azure/cluster-api-azure-controller:v1.3.1` should be available.

**Anything else you would like to add:**

**Environment:**

- cluster-api-provider-azure version: [v1.3.1](https://github.com/kubernetes-sigs/cluster-api-provider-azure/releases/tag/v1.3.1)
- Kubernetes version: (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6", GitCommit:"ad3338546da947756e8a88aa6822e9c11e7eac22", GitTreeState:"clean", BuildDate:"2022-04-14T08:49:13Z", GoVersion:"go1.17.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0", GitCommit:"4ce5a8954017644c5420bae81d72b09b735c21f0", GitTreeState:"clean", BuildDate:"2022-05-10T01:37:08Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
```
- OS (e.g. from `/etc/os-release`): 
```
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

**Comment by jsturtevant** [comment]
We missed the publishing of the image, @mboersma updated it this morning: https://github.com/kubernetes/k8s.io/pull/3745

**Comment by ionutbalutoiu** [comment]
It seems that the image is available now.

It caused the above issue, because it was not available with the release of `v1.3.0`.
